### PR TITLE
Post only non-null values in request body for createTree

### PR DESCRIPTION
**Pull request by asthinasthi**
# Description 
Github release 2.18+ doesn't accept `sha:null` in the request body. This change fixes that. More details here: https://github.com/github-api/github-api/issues/636

# Before submitting a PR:
We love getting PRs, but we hate asking people for the same basic changes every time. 

- [x] Push your changes to a branch other than `master`. Create your PR from that branch.    
- [ ] Add JavaDocs and other comments
- [x] Write tests that run and pass in CI. See [CONTRIBUTING.md](CONTRIBUTING.md) for details on how to capture snapshot data.
- [x] Run `mvn install site` locally. This may reformat your code, commit those changes. If this command doesn't succeed, your change will not pass CI.

### CODE REVIEW

**Comment by bitwiseman** [comment]
@asthinasthi 
This is really good.  Thanks!  This is the behavior for GHE.  Is it the same for GitHub.com?

**Comment by asthinasthi** [comment]
Actually I see that we have deprecated this https://github.com/github-api/github-api/blob/d1507f26668950508e0bf242c34cdb599003991a/src/main/java/org/kohsuke/github/GHTreeBuilder.java#L111
in the latest release (I was using 1.93).

So when we use the recommended 
https://github.com/github-api/github-api/blob/d1507f26668950508e0bf242c34cdb599003991a/src/main/java/org/kohsuke/github/GHTreeBuilder.java#L129

```

    public GHTreeBuilder add(String path, byte[] content, boolean executable) {
        try {
            String dataSha = repo.createBlob().binaryContent(content).create().getSha();
            return shaEntry(path, dataSha, executable);
        } catch (IOException e) {
            throw new GHException("Cannot create binary content of '" + path + "'", e);
        }
    }
```
We create a sha & then use it in the API. We never send a null sha field.

Also, we have deprecated `shaEntry(path, dataSha, executable)` so I think we are good.

**Comment by bitwiseman** [comment]
@asthinasthi 
Um, so, do you want to close this or still merge it?

**Comment by asthinasthi** [comment]
We can close this. As long as I stay away from deprecated methods I should be fine.

**Comment by Xcelled** [comment]
Can this be reopened and merged? I just got bit by this issue in deployed production code after a GH:E upgrade and it took me forever to track it down.

File: src/main/java/org/kohsuke/github/GHTreeBuilder.java [hunk 1]
```diff
@@ -16,8 +18,10 @@
 
     private final List<TreeEntry> treeEntries = new ArrayList<TreeEntry>();
 
+    @JsonInclude(Include.NON_NULL)
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHTreeBuilder.java]:
```suggestion
    // Issue #636: Create Tree no longer accepts null value in sha field
    @JsonInclude(Include.NON_NULL)
```

**Pull request merged**